wicketd/src/http_entrypoints.rs

```diff
@@ -1237,6 +1241,51 @@ async fn get_preflight_uplink_report(
     }
 }
 
+/// An endpoint instructing wicketd to reload its SMF config properties.
+///
+/// The only expected client of this endpoint is `curl` from wicketd's SMF
+/// `refresh` method, but other clients hitting it is harmless.
+#[endpoint {
+    method = POST,
+    path = "/reload-config",
+}]
+async fn post_reload_config(
+    rqctx: RequestContext<ServerContext>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let smf_values = SmfConfigValues::read_current().map_err(|err| {
+        HttpError::for_unavail(
+            None,
+            format!("failed to read SMF values: {err}"),
+        )
+    })?;
+
+    let Some(smf_values) = smf_values else {
+        return Err(HttpError::for_bad_request(
+            None,
+            "reloading config from SMF only available on illumos".to_string(),
```

Comment:
Oh, so on non-illumos platforms we'd expect it to `SmfConfigValues::read_current()` to return `None` rather than throwing an error?